openapi3/discriminator.go

```diff
@@ -10,8 +10,22 @@ import (
 type Discriminator struct {
 	Extensions map[string]any `json:"-" yaml:"-"`
 
-	PropertyName string    `json:"propertyName" yaml:"propertyName"` // required
-	Mapping      StringMap `json:"mapping,omitempty" yaml:"mapping,omitempty"`
+	PropertyName string                `json:"propertyName" yaml:"propertyName"` // required
+	Mapping      map[string]MappingRef `json:"mapping,omitempty" yaml:"mapping,omitempty"`
```

Comment:
Please rebase. Sorry about that, I noticed your issue and PR after merging the other one.

Also please keep the newtype `StringMap` functionality (see recent changes).

Reply:
Sorry I did write this before the `StringMap` change came in and naïvely rebased, I saw this patch as a proposal for discussion but if you believe this is the right approach then I will fix it up

Reply:
To build upon the existing `StringMap` I have introduced a change to make it accept a generic value. It is only used in 2 places so the diff is tiny